yamllint/cli.py

```diff
@@ -140,6 +140,18 @@ def show_problems(problems, file, args_format, no_warn):
 
     return max_level
 
+def find_project_config(path='.'):
+
+    for filename in [ '.yamllint', '.yamllint.yaml', '.yamllint.yml' ]:
+        file = os.path.join(path, filename)
+        if os.path.isfile(file):
+            return file
+
+    if os.path.abspath(path) == os.path.abspath(os.path.expanduser('~')):
+        return False
+    if os.path.abspath(path) == os.path.abspath('/'):
+        return False
```

Comment:
- I suggest returning `None` rather than `False`, this is more Pythonic.
- Will the comparison to `os.path.abspath('/')` work on Windows-like systems?

Reply:
> Will the comparison to `os.path.abspath('/')` work on Windows-like systems?

It seems that it will work...

```
Python 3.10.8 (tags/v3.10.8:aaaf517, Oct 11 2022, 16:50:30) [MSC v.1933 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> os.getcwd()
'C:\\Program Files\\Python310'
>>> os.path.abspath(os.path.join(os.getcwd(), '..', '..'))
'C:\\'
>>> os.path.abspath('/')
'C:\\'
>>> os.path.abspath(os.path.join(os.getcwd(), '..', '..')) == os.path.abspath('/')
True
```

Reply:
I addressed all your comments and tried running the tests on Windows, but unfortunately there are too many errors on Windows.

First try:

```
python -m unittest tests/test_cli.py
E
======================================================================
ERROR: test_cli (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: test_cli
Traceback (most recent call last):
  File "C:\Program Files\Python310\lib\unittest\loader.py", line 154, in loadTestsFromName
    module = __import__(module_name)
  File "C:\Users\georgi.georgiev\work\git\yamllint\tests\test_cli.py", line 17, in <module>
    import fcntl
ModuleNotFoundError: No module named 'fcntl'


----------------------------------------------------------------------
Ran 1 test in 0.000s

FAILED (errors=1)
```

After deleting the impossible tests, I am getting many errors mostly due to new line format and path separators...

Small sample:
```
First differing element 0:
1
0

- (1,
+ (0,
-  '.\\a.yml:1:1: [warning] missing document start "---" (document-start)\n'
?    ^^

+  './a.yml:1:1: [warning] missing document start "---" (document-start)\n',
?    ^                                                                     +

-  '.\\a.yml:1:13: [error] wrong new line character: expected \\n (new-lines)\n',
   '')
```

```
First differing element 5:
'C:\\[60 chars]3teckb\\s\\s\\s\\s\\s\\s\\s\\s\\s\\s\\s\\s\\s\\s\\s\\file.yaml'
'C:\\[60 chars]3teckb\\s/s/s/s/s/s/s/s/s/s/s/s/s/s/s/file.yaml'
```

But I think the comparison is good, I even ran yamllint on windows and it doesn't trigger a recursion limit.